recirq/seniority_zero/data_processing/general.py

```diff
@@ -0,0 +1,426 @@
+# Copyright 2023 Google
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""General data processing functions"""
+from typing import Dict, List, Optional, Union
+
+import cirq
+import numpy as np
+import pandas
+import scipy
+from openfermion import QubitOperator
+
+from recirq.seniority_zero.scheduling import SnakeMeasurementGroup
+
+
+def multi_measurement_histogram(result: Union[cirq.Result, pandas.DataFrame], keys: List[str]):
+    """Converts either a cirq.Result or pandas.Dataframe to a histogram
+
+    ####### ASSUMES EACH KEY IS ASSIGNED TO A MEASUREMENT WITH 1 RESULT #######
+    """
+    if isinstance(result, cirq.Result):
+        return result.multi_measurement_histogram(keys=keys)
+
+    import collections
+
+    from cirq.study.result import _key_to_str, _tuple_of_big_endian_int
+    from cirq.value import big_endian_int_to_bits
```

Comment:
can we import these at the top level of the module?

Reply:
Fixed - I think this was copied out of cirq and changed a bit lol.